### DISCUSSION TITLE
webhook on local machine

### DISCUSSION THREAD

**Discussion by mramitdas** [FabCx/TradingView-Webhook-Bot#39] [General]
Hi there,

i am strugelling while trying to get the webhook on my local machine.

1. what should i put in webhook url (Trading view Website
![Screenshot (49)](https://user-images.githubusercontent.com/51442266/112799768-6f2d6800-908c-11eb-85a9-f012354e3b7e.png)
)

2. which ip you are talking about ( ipv4 (192.168.**.**) or internet ip (45. **. **. **) )

Thanks in advance☺

**Comment by fabston** [comment]
Hi, 

For outside communication you always have to use your WAN IP. Which seems to be for you 45.xx.xx.

You most likely have to also open ports on your router. Be sure to keep that in mind. For stability reason I always recommend to use a VPS such as [Hetzner](https://fabston.dev/hetzner).

**Reply by mramitdas** [reply]
i tried to do it but it gives me error..
![Screenshot (50)](https://user-images.githubusercontent.com/51442266/112804572-342e3300-9092-11eb-96c0-f81907490cb1.png)

**Reply by fabston** [reply]
Ports need to be open inside your router interface. You don't have to change anything inside the bot since it will remain to run on localhost.

**Reply by mramitdas** [reply]
Please guide me how to achieve that
![Screenshot (52)](https://user-images.githubusercontent.com/51442266/112816955-8d509380-909f-11eb-9b64-f0633db50dc9.png)

**Reply by fabston** [reply]
You could whitelist TradingViews Webhook IPs using port 80. You can find the IPs [here](https://www.tradingview.com/support/solutions/43000529348-about-webhooks/). Not sure how your Router Interface works so you would have to do a bit of digging yourself.

**Reply by mramitdas** [reply]
can i host this on heroku ? if yes please share the process

**Reply by fabston** [reply]
I guess you can, but I never tried it. As mentioned before, it's recommended to run it on a VPS like Vultr, Hetzner etc.

**Reply by mramitdas** [reply]
can you please guide me for either local machine or my server. i also have owned a domain specially for this purpose. please guide me. thanks in advance.

**Reply by fabston** [reply]
I've mentioned each step in the [Installation](https://github.com/fabston/TradingView-Webhook-Bot#installation) section. A domain is not really required, however it can be used to run it behind a SSL cert.

**Reply by mramitdas** [reply]
Thanks for the guidence. i configured my router. but another issue came up my wan ip is different from public ip which is not accepted by tradingview..

![Screenshot (53)](https://user-images.githubusercontent.com/51442266/113263246-288b7800-92ef-11eb-9aba-c819b071c7e1.png)
![Screenshot (54)](https://user-images.githubusercontent.com/51442266/113263253-29bca500-92ef-11eb-8289-1feeaa3e8eef.png)

**Reply by fabston** [reply]
I can't further assist you on setting up the bot on a local network, since I'm not running it on one either. As mentioned in the readme its recommended to use a VPS instead.

**Reply by fabston** [reply]
Also keep in mind your IP can change daily depending on your ISP. Overall not recommended to run it on your local network, only if you know what your doing.

**Reply by mramitdas** [reply]
Thanks.. Currently working with heroku.. Will update you soon. Thanks for the help 😊

**Reply by mramitdas** [reply]
Hi there,
I have successfully implemented the code on heroku. If like to know i will share everything.

**Reply by fabston** [reply]
Cool! Feel free to share the code or do a pull request. Appreciate it.
